src/main/java/bit/javaoop/FixedBudgetManager.java

```diff
@@ -0,0 +1,35 @@
+package bit.javaoop;
+
+import java.util.ArrayList;
+
+public class FixedBudgetManager extends AbstractManager {
+
+    private double budget;
+
+    public FixedBudgetManager(String name, double salary, ArrayList<AbstractEmployee> employeesList, double budget) {
+        super(name, salary, employeesList);
+        this.budget = budget;
+    }
+
+    @Override
+    public boolean isSatisfied() {
+        return (getSalary() > 20_000) && (budget < 15_000);
+    }
+
+    @Override
+    public boolean canHire() {
+        double employeesSalaries = 0;
+        for (AbstractEmployee employee : employeesList) {
+            employeesSalaries += employee.getSalary();
+        }
+        return (budget >= employeesSalaries);
+    }
+
+    @Override
+    public void hire(AbstractEmployee employee) {
```

Comment:
Metoda bazowa mogłaby się nazywać `hireIfPossible`, gdyż jeśli nie jest w stanie kogoś zatrudnić to nie robi po prostu nic. Taka nazwa lepiej będzie opisywać działanie tej metodki.